src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/ConfirmServiceCallbackHandler.java

```diff
@@ -119,34 +118,29 @@ private CallbackResponse prepareCertificateOfService(CallbackParams callbackPara
         } else {
             serviceDate = caseData.getServiceDateTimeToRespondentSolicitor1();
         }
-        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
-
         LocalDate deemedDateOfService = deadlinesCalculator.calculateDeemedDateOfService(
             serviceDate, serviceMethod.getType());
         LocalDateTime responseDeadline = deadlinesCalculator.calculateDefendantResponseDeadline(deemedDateOfService);
 
-        data.put("deemedServiceDateToRespondentSolicitor1", deemedDateOfService);
-        data.put("respondentSolicitor1ResponseDeadline", responseDeadline);
-
-        CaseData caseDateUpdated = caseData.toBuilder()
+        CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder()
             .deemedServiceDateToRespondentSolicitor1(deemedDateOfService)
-            .respondentSolicitor1ResponseDeadline(responseDeadline)
-            .build();
+            .respondentSolicitor1ResponseDeadline(responseDeadline);
 
         CaseDocument certificateOfService = certificateOfServiceGenerator.generate(
-            caseDateUpdated,
+            caseDataBuilder.build(),
```

Comment:
really don't like the requirement to build case data here, is hard to know what is needed for generate method. Again - maybe not your PRs concern 😅

Reply:
will visit this later if needed.